virtualizarr/manifests/group.py

```diff
@@ -0,0 +1,39 @@
+from typing import TypeAlias
+
+from zarr.core.group import GroupMetadata
+
+from virtualizarr.manifests import ManifestArray
+
+ManifestArrayVariableMapping: TypeAlias = dict[str, ManifestArray]
+
+
+class ManifestGroup:
+    """
+    Virtualized representation of multiple ManifestArrays as a Zarr Group.
+    """
+
+    # TODO: Consider refactoring according to https://github.com/zarr-developers/VirtualiZarr/pull/490#discussion_r2007805272
+    _manifest_arrays: ManifestArrayVariableMapping
```

Comment:
Why not just 
```suggestion
    _manifest_arrays: Mapping[str, ManifestArray]
```
The actual implementation can still use a dict, but this choice of typing clearly communicated that the `Mapping` is not meant to be mutable once set (because the `ManifestStore` is not meant to be writable).